src/components/shared/Visualizer.vue

```diff
@@ -187,6 +197,20 @@ export default {
         };
       });
     },
+    addSizingListeners() {
+      const wrapper = this.$el;
+      this.svgWidth = wrapper.clientWidth;
+      window.onresize = this.handleSVGParentResize;
+    },
+    handleSVGParentResize() {
+      const wrapper = this.$el;
+      this.svgWidth = wrapper.clientWidth;
```

Comment:
Since the `handleSVGParentResize()` will be called a lot, you can try to optimize the use of memory within the operation.
`this.svgWidth = this.$el.clientWidth;`

Reply:
I don't agree that it will be called a lot, not many people toy around with the window size, I also think that the removal of the variable will not make it less readable for an expert vue developer, but it can help less experienced ones. Let me know what you think.

Reply:
True about helping less experienced, but comments there for that.
`// Store the width of the wrapper
this.svgWidth = this.$el.clientWidth;`
And about resizing is true, but our tree can start getting very big, so maybe user will zoom or play with the size to get better view of the tree